editoast/src/views/timetable.rs

```diff
@@ -933,4 +986,80 @@ mod tests {
             .json_into();
         assert_eq!(&response.error_type, "editoast:timetable:NotFound")
     }
+
+    #[rstest]
+    #[case(Some(1))]
+    #[case(None)]
+    fn train_id_parse_paced_train_exception(#[case] index: Option<u64>) {
+        let exception_key = "5062360642-f8f8-44ef-9f37-22cd0f5fcc07".to_string();
+        let paced_train_id = 50;
+
+        let paced_train_exception_id = TrainId::PacedTrainException {
+            paced_train_id,
+            exception_key: exception_key.clone(),
+            index,
+        }
+        .to_string()
+        .parse::<TrainId>()
+        .unwrap();
+        assert_eq!(
+            paced_train_exception_id,
+            TrainId::PacedTrainException {
+                paced_train_id,
+                exception_key: exception_key.clone(),
+                index
+            }
+        );
+    }
+
+    #[test]
+    fn train_id_parse_paced_train_occurrence() {
+        let index = 1;
+        let paced_train_id = 50;
+        let paced_train_occurence_id = TrainId::PacedTrainOccurrence {
+            paced_train_id,
+            index,
+        }
+        .to_string()
+        .parse::<TrainId>()
+        .unwrap();
+
+        assert_eq!(
+            paced_train_occurence_id,
+            TrainId::PacedTrainOccurrence {
+                paced_train_id,
+                index
+            }
+        );
+    }
+
+    #[test]
+    fn train_id_parse_train_schedule() {
+        assert_eq!(
+            TrainId::TrainSchedule(30)
+                .to_string()
+                .parse::<TrainId>()
+                .unwrap(),
+            TrainId::TrainSchedule(30)
+        );
+    }
```

Comment:
I believe you can compact these 4 tests into a much shorter one. What you care about if I understand correctly is that you can do a full roundtrip of converting to string then parse, to verify that the input is identical to the output. But if you start by parsing, then convert to string, you get the same benefit, but it can be written in a much shorter way.

```rust

    #[rstest]
    #[case("42")]
    #[case("42#10")]
    #[case("84@exception_21")]
    #[case("84@exception_21#7")]
    fn train_id_parse_and_to_string_roundtrip(#[case] id: &str) {
        assert_eq!(id.parse::<TrainId>().unwrap().to_string(), id);
    }
```